web-marketplace/src/components/organisms/PostForm/PostForm.tsx

```diff
@@ -0,0 +1,212 @@
+    if (files?.[fieldIndex]?.['schema:url'] === DEFAULT_URL) {
+      append({
+        'schema:url': DEFAULT_URL,
+      });
+    }
+    setValue(`files.${fieldIndex}.schema:url`, encodeURI(value));
+  };
+
+  const getHandleDeleteWithIndex = async (fieldIndex: number) => {
+    remove(fieldIndex);
+  };
+
+  /* Effect */
+
+  useEffect(() => {
+    if (
+      fields?.length === 0 ||
+      fields?.every(field => field['schema:url'] !== DEFAULT_URL)
+    ) {
+      append({
+        'schema:url': DEFAULT_URL,
+      });
+    }
+  }, [append, fields]);
+
+  return (
+    <Form className={cn('max-w-[560px]', className)} form={form}>
+      <Title variant="h3" sx={{ textAlign: 'center' }} className="mb-50">
+        New post
+      </Title>
+      <TextField
+        type="text"
+        label="Title"
+        description="Summarize this update."
+        className="mb-50"
+        helperText={
+          <TextAreaFieldChartCounter
+            value={title}
+            charLimit={POST_MAX_TITLE_LENGTH}
+            sx={{ mb: { xs: 0, sm: 0 } }}
+          />
+        }
+        {...form.register('title')}
+      />
+      <TextAreaField
+        type="text"
+        label="Comment"
+        description="Write a short comment or longer project update. "
+        rows={4}
+        minRows={4}
+        multiline
+        className="mb-50 mt-0"
+        {...form.register('comment')}
+      />
+      {fields.map((field, index) => {
+        const url = files?.[index]?.['schema:url'];
+
+        return (
+          <ImageDrop
+            label={'Files'}
+            description={
+              <Body>
+                {
+                  '5MB max. Supported file types include text, spreadsheets, images, and video files. '
+                }
+                <Link href="#">{'View all supported file types»'}</Link>
+              </Body>
+            }
+            onDelete={() => getHandleDeleteWithIndex(index)}
+            value={url === DEFAULT_URL ? '' : url}
+            setValue={setFiles}
+            className={cn('mb-50 mt-0', classes.galleryItem)}
+            key={field.id}
+            fieldIndex={index}
+            error={!!errors['files']}
+            helperText={errors['files']?.message}
+            renderModal={() => <div />}
+            optional
+            {...imageDropCommonProps}
+            {...form.register('files')}
+          />
+        );
+      })}
+      <div className="flex flex-col mb-50">
+        <RadioGroup
+          label={
+            <span className="inline-flex items-center">
+              <LockIcon className="mr-10" />
+              {'Privacy settings'}
+            </span>
+          }
+          description={POST_PRIVACY_DESCRIPTION}
+          optional
+        >
+          <>
+            <Radio
+              label={
+                <span className={cn('flex items-center')}>
+                  <PostIcon className="mr-15" />
+                  {'Make the entire post private'}
```

Comment:
this should only be bold when selected:

![image](https://github.com/regen-network/regen-web/assets/3149358/b2c5bb11-698d-4f13-9c2c-bea9fbf1fa06)